security-privacy-self-review.md

```diff
@@ -0,0 +1,78 @@
+
+
+Responses to the [Self-Review Questionnaire: Security and Privacy](https://w3ctag.github.io/security-questionnaire/) for the [Digital Goods API](https://github.com/WICG/digital-goods/)
+
+
+## 2.1 What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+The API allows an origin to query the details of digital goods available from a store backend, for that origin and user.
+It exposes the user's chosen currency (as part of the price details) and which items have been purchased.
+This information is needed for the origin to display purchases to the user and give the user access to features/purchases based on purchases made.
+
+The API also allows an origin to know that a purchase has been acknowledged, which is necessary to confirm that a transaction has completed.
+
+## 2.2 Is this specification exposing the minimum amount of information necessary to power the feature?
+
+Yes.
+
+## 2.3 How does this specification deal with personal information or personally-identifiable information or information derived thereof?
+
+The API involves an already-authenticated user and their purchases from that origin, but doesn't expose any other information about the user. No information is persisted.
```

Comment:
Hm, so we keep saying "already-authenticated" but that's already authenticated _with the browser_. There's no guarantee that they would be authenticated with the website, is there? So this API could allow a website that otherwise has no way of knowing who the user is, get access to some identifying information, right?

In the Google use case, they would have to have been installed through the Play Store, so I don't think it presents an increased risk in that context (since Play apps have access to the user's ID), but as a general thing, maybe we should call out that implementors should consider exposure of this information.

Reply:
Good point. We're potentially revealing information in the case that the user is authenticated with the browser but not with the site. I'll add a note about this case.

Is there anything we can suggest to mitigate the issue? GetDetails and Acknowledge don't reveal much identifying information (just currency locale, AFAIK), but ListPurchases could identify a specific user (if it includes accurate purchase timestamps).